sycl/include/sycl/types.hpp

```diff
@@ -2401,6 +2402,35 @@ struct is_device_copyable<
 template <typename T>
 inline constexpr bool is_device_copyable_v = is_device_copyable<T>::value;
 
+// std::array<T, 0> is implicitly device copyable type.
+template <typename T>
+struct is_device_copyable<std::array<T, 0>> : std::true_type {};
+
+// std::array<T, N> is implicitly device copyable type if T is device copyable
+// and it is not trivially copyable (if it is trivially copyable it is device
```

Comment:
Why do we need this extra condition on "not trivially copyable" here? Why can't we just say that `array<T, N>` is trivially copyable if `T` is trivially copyable?

Reply:
We need this extra condition because if it is trivially copyable there are two candidates and the compiler can't decide.